### DISCUSSION TITLE
The best way to persist mobx-store without decorators?

### DISCUSSION THREAD

**Discussion by caiphaav** [mobxjs/mobx#2598] [Q&A]
Can somebody provide an example how to correctly persist store in Mobx 6 without decorators?

**Comment by kubk** [comment]
It's documented here, search for "without decorators" string: https://github.com/pinqy520/mobx-persist
An example: https://github.com/pinqy520/mobx-persist/blob/master/dev/non-decorator.tsx

**Reply by caiphaav** [reply]
I'm trying to use it with mobx 6 but, unfortunately , it doesn't persist any data

**Comment by danielkcz** [comment]
The `mobx-persist` is 3 years old, so chances are it's not indeed compatible with V6. But you can look at its source code and find inspiration on how to do it in your case.

Depending on how far are you in development, you can consider mobx-state-tree or mobx-keystone for your stores instead. Both support snapshots and patches which can be used for persistence nicely.
